intake/cli/server/server.py

```diff
@@ -353,6 +353,8 @@ def post(self):
             msg = dict(format=chunk_encoder.format_name,
                        compression=chunk_encoder.compressor_name,
                        container=source.container, data=data)
+            # Is this double serializing / copying?
+            # We're passing a Buffer object here.
```

Comment:
I could use some help understanding the flow here. At this point, I *think* data has already been serialized. I got an error about msgpack not being able to serialize a pyarrow Buffer object (which I worked around by wrapping it in a memoryview).

But that makes me think that we've already serialized the data, and are potentially serializing it again? Does that make sense?

Reply:
Right, we are serialising a few times: dataframe to bytes (arrow buffer, which is just a buffer), then to msgpack (which will contain that bytestring plus other stuff) and then to JSON to send over the connection. The latter two steps don't require much processing, but will be new memory copies.